source/Engine.cpp

```diff
@@ -1776,7 +1776,9 @@ void Engine::SendHails()
 		return;
 
 	// Generate a random hail message.
-	SendMessage(source, source->GetHail(player));
+	map<string, string> stubs;
+	player.FillHailSubs(stubs);
+	SendMessage(source, source->GetHail(stubs));
```

Comment:
Since no one needs to do it yet, use the more idiomatic signature:
```suggestion
	SendMessage(source, source->GetHail(player.GetSubstitutions());
```
where the PlayerInfo method returns the map by value, rather than the C-style API.

When someone needs to add the player subs to their own subs, and they can't start from the value returned by this method, then we can add the `void PlayerInfo::AddSubstitutions(map<string, string> &) const` method.

Reply:
Done